### PR TITLE
docs(clone): add promise example

### PR DESCRIPTION
**Pull request by vvo**
It took me some time to figure out how to properly create a pipeline and know
when it's done, while https://github.com/lovell/sharp/issues/524 provides an
example about listening to the finish event, I needed one for toFile.

Now I think we should also document the example from
https://github.com/lovell/sharp/issues/524 but I'd like you the maintainer to
create it because I am not 100% sure this is the most elegant way.

Thanks for the awesome library! It helped me build this: https://coronamaison.net/ (https://github.com/vvo/coronamaison/blob/549a807aea1e2034a0a5821dff6ea89488be46a3/prepareData.js#L153)

### CODE REVIEW

File: lib/constructor.js [hunk 1]
```diff
@@ -253,6 +253,56 @@ util.inherits(Sharp, stream.Duplex);
...
+ *     .clone()
+ *     .resize({ width: 500 })
+ *     .webp({ quality: 80 })
+ *     .toFile("optimized-500.webp")
+ * );
+ *
+ * // https://github.com/sindresorhus/got#gotstreamurl-options
+ * got.stream("https://www.example.com/some-file.jpg").pipe(sharpStream);
+ *
+ * fs.createReadStream(jpgOriginalImagePath).pipe(sharpStream);
```
Comment (lovell) [inline lib/constructor.js:293]: Might the piping of two streams into the same sharp instance confuse people? I'd prefer to stick to one example, perhaps focus on the use of `got` as the use of HTTP Stream-based input is more common (for filesystem-based input, passing the path to the constructor is preferable).

Comment (vvo) [reply lib/constructor.js:293]: ah! that's a mistake of me, in my code those two (got/fs.createReadStream) are in different code paths (if ...), Ill update that! (definitely confusing lol)

Comment (vvo) [reply lib/constructor.js]: ok just updated the pr now, thx!

**Comment by lovell** [comment]
Landed in commit 1eedb22 - merci beaucoup!

**Pull request closed without merging**